### PR TITLE
(AIP-44) Migrate TaskCommand._get_ti to Internal API

### LINKED ISSUE #29320: AIP-44 Migrate TaskCommand._get_ti to Internal API
https://github.com/apache/airflow/blob/main/airflow/cli/commands/task_command.py#L145

### PR DESCRIPTION
**Pull request by mhenc**
Change TaskCommand._get_ti to return either TaskInstance or TaskInstancePydantic, making sure it calls the InternalAPI in dbless mode.

Some CLI commands, despite calling TaskCommand._get_ti still assume access to database.

closes: #29320
related: #29320

### CODE REVIEW

**Comment by mhenc** [comment]
cc: @potiuk @vincbeck

File: airflow/api/common/experimental/get_task_instance.py [hunk 1]
```diff
@@ -42,5 +42,6 @@ def get_task_instance(dag_id: str, task_id: str, execution_date: datetime) -> Ta
     if not task_instance:
         error_message = f"Task {task_id} instance for date {execution_date} not found"
         raise TaskInstanceNotFound(error_message)
-
+    # API methods has access to the database.
+    assert isinstance(task_instance, TaskInstance)
```
Comment (hussein-awala) [inline airflow/api/common/experimental/get_task_instance.py]: assert statement is ignored in runtime; if you want to check this in runtime, you should switch to if...raise

File: airflow/models/dagrun.py [hunk 2]
```diff
@@ -466,7 +467,7 @@ def generate_run_id(run_type: DagRunType, execution_date: datetime) -> str:
     def fetch_task_instances(
         dag_id: str | None = None,
         run_id: str | None = None,
-        dag: DAG | None = None,
+        task_ids: list[str] | None = None,
```
Comment (hussein-awala) [inline airflow/models/dagrun.py:470]: This is a breaking change; we cannot remove an argument from a public method. Could you deprecate it instead, or even keep it and support both params?

File: airflow/www/views.py [hunk 3]
```diff
@@ -1543,6 +1543,7 @@ def rendered_k8s(self, *, session: Session = NEW_SESSION):
 
         pod_spec = None
         try:
+            assert isinstance(ti, TaskInstance)
```
Comment (hussein-awala) [inline airflow/www/views.py]: same here for assert.

File: airflow/models/dagrun.py (hunk 2 above)
Comment (vincbeck) [reply airflow/models/dagrun.py:470]: Unless I am mistaken this is not breaking change. I think we can change signature of public methods as long as they are not [public interface](https://airflow.apache.org/docs/apache-airflow/stable/public-airflow-interface.html). I dont expect users to use this method in their DAGs so I think it is safe to modify it. Not being able to modify any public method would restraint very much the development in Airflow IMHO

File: airflow/api/common/experimental/get_task_instance.py (hunk 1 above)
Comment (mhenc) [reply airflow/api/common/experimental/get_task_instance.py]: Fair point - switched to if-raise as we won't plan to change it in the near future.

**Comment by vincbeck** [comment]
There are some circular dependency issues

File: airflow/models/dagrun.py (hunk 2 above)
Comment (hussein-awala) [reply airflow/models/dagrun.py:470]: That's right!

**Review by hussein-awala — APPROVED** [review]
Looks good

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Some tests are failing though

**Review by vincbeck — APPROVED** [review]

**Comment by potiuk** [comment]
BTW. Fix to failing "Skip Build" in https://github.com/apache/airflow/pull/35503

**Comment by mhenc** [comment]
Finally I made the tests green. There were some changes required. PTAL.

**Review by vincbeck — APPROVED** [review]

**Pull request merged**